glean_parser/templates/rust.jinja2

```diff
@@ -150,14 +150,27 @@ pub mod {{ category.name|snake_case }} {
         {%- for arg_name in extra_metric_args if not obj.labeled and obj[arg_name] is defined and arg_name != 'allowed_extra_keys' -%}
             , {{ obj[arg_name]|rust }}
         {%- endfor -%}
-        {{ ", " if obj.labeled else ")\n" }}
+        {{ ", " if obj.labeled or obj.dual_labeled else ")\n" }}
         {%- if obj.labeled -%}
         {%- if obj.labels -%}
         Some({{ obj.labels|rust }})
         {%- else -%}
         None
         {%- endif -%})
         {% endif %}
+        {%- if obj.dual_labeled -%}
+        {%- if obj.keys -%}
+        Some({{ obj.keys|rust }})
+        {%- else -%}
+        None
+        {%- endif -%}
```

Comment:
musing: Feels like the "if it's python!`None` then rust!`None`, otherwise pass it through the filter" could be made a feature of the rust type transformation function.